core/controllers/suggestion.py

```diff
@@ -97,12 +97,20 @@ def get(self):
         if not constants.USE_NEW_SUGGESTION_FRAMEWORK:
             raise self.PageNotFoundException
 
-        queries = self.request.GET.items()
-        for query in queries:
+        # The query_fields_and_values variable is a list of tuples. The first
+        # element in each tuple is the field being queried and the second
+        # element is the value of the field being queried.
+        # request.GET.items() parses the params from the url into the above
+        # format. So in the url, the query should be passed as:
+        # ?field1=value1&field2=value2...fieldN=valueN.
+        query_fields_and_values = self.request.GET.items()
```

Comment:
OK, one last thing -- can you add some tests to check escaping/unescaping? E.g. suppose the field value contained an ampersand, or unicode chars, etc.

Reply:
You mean parsing of the field and value from the url? If so, this feels like a more general test for the request.GET.items() function right? Shall I write one in a different file (say controllers/base_test.py)

Reply:
Yeah, I think that's fine too. It's the first time we've used GET.items(), I think, so we should check that we can get the stuff properly from it.

That said, if you do need to do a transformation to the return value of GET.items() to get the correct data, a local test here probably makes more sense.

Reply:
Yeah. As per docs it gives list(tuples(field, value)) so we don't need to do any transformations and just pass the output directly. I'll try some corner cases on the GET.items() function in a separate test suite, like the & and unicode characters as you said.

Reply:
Hi @seanlip, I thought about this situation. So what I feel is that these characters don't stay the same in the URL right? so & becomes %26, <space> becomes + and so on. So I am not sure of what to test here. Any suggestions as to what I should test? 
Thanks!